AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/HierarchicalVersionUtils.dfy

```diff
@@ -64,4 +66,102 @@ module HierarchicalVersionUtils {
   {
     (bkcDigest + aes256Key)
   }
+
+  // HV-2 only sends the Branch Key's Encryption Context to KMS, as compared to the Branch Key Context
+  // the Branch Key's Encryption Context can be divided into two groups
+  // 1. those created by MPL consumers via the library, which are prefixed with aws-crypto-ec:
+  // 2. those created by MPL consumers OUTSIDE of the library, which are not prefixed
+  function SelectKmsEncryptionContextForHv2(
+    branchKeyContext: Types.EncryptionContextString
+  ): (output: Types.EncryptionContextString)
+    requires Structure.BranchKeyContext?(branchKeyContext)
+    // TODO-HV-2-M2: Revisit this implementation to handle scenarios where removing prefix results in conflicting keys.
+    requires forall k1, k2 <- branchKeyContext.Keys ::
+               (if HasPrefix(k1) then RemovePrefix(k1) else k1) ==
+               (if HasPrefix(k2) then RemovePrefix(k2) else k2) ==>
+                 k1 == k2
+    ensures forall k <- output ::
+              exists originalKey ::
+                && originalKey in branchKeyContext
+                && (HasPrefix(originalKey) ==> k == RemovePrefix(originalKey))
+                && (!HasPrefix(originalKey) ==> k == originalKey)
+                && originalKey !in Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES
+  {
+    var transformedContext :=
+      set k <- branchKeyContext.Keys
+          | k !in Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES
+        :: (
+             if HasPrefix(k) then RemovePrefix(k) else k,  // transformed key
+             branchKeyContext[k],                          // original value
+             k                                             // original key
```

Comment:
Non blocking: We could remove original key from the tuple. Its not used any where